### DISCUSSION TITLE
Will the WinRT APIs be upgraded to minimize the use of legacy Win 32 APIs?

### DISCUSSION THREAD

**Discussion by themerror** [microsoft/WindowsAppSDK#2678] [Ideas]
For example, when putting the application on the top level, we need to implement it through the following code:
`SetWindowPos(m_hWndTop,HWND_TOPMOST,0,0,0,0,SWP_NOMOVE|SWP_NOSIZE);`
Will there be some new WinRT APIs to use (`SetCurrentWindowTopMost()` or something like this) and permissions associated with it(`windowTopMost`)?

**Comment by ghost1372** [comment]
For TopMost you can use AppWindow

**Reply by themerror** [reply]
I find some problems when I tried to use it as [Show multiple views with AppWindow](https://docs.microsoft.com/en-us/windows/apps/design/layout/app-window) said
```
AppWindow appWindow = await AppWindow.TryCreateAsync();
Frame appWindowContentFrame = new Frame();
appWindowContentFrame.Navigate(typeof(AppWindowMainPage));
ElementCompositionPreview.SetAppWindowContent(appWindow, appWindowContentFrame);
await appWindow.TryShowAsync();
```
In Windows App SDK, it may need to be changed to 
```
AppWindow appWindow = AppWindow.Create();
Frame appWindowContentFrame = new Frame();
appWindowContentFrame.Navigate(typeof(MainPage1));
ElementCompositionPreview.SetAppWindowContent(appWindow, appWindowContentFrame);
appWindow.Show();
``` 
But `ElementCompositionPreview.SetAppWindowContent(appWindow, appWindowContentFrame);` hasn't been a part of WASDK yet, so how can I fix this?
BTW, I hope Microsoft could improve their documents. It's really hard to find out what can be used in WASDK and what can only be used in UWP apps or what should be changed so we can use them in WASDK.
Thanks.

**Reply by themerror** [reply]
And please forgive me for using a not-so-good example, but in fact what I want to say is whether Microsoft will continue to improve the WinRT API to ensure that developers can minimize the direct use of the Win 32 API when using WASDK or UWP (if not abandoned), and to make the application as controllable and secure when calling the WinRT API as UWP or other operating systems with containers (such as Android).
